### PR TITLE
Committed range iterator

### PR DESCRIPTION
**Pull request by nopcoder**
- committed.Iterator will use RangeManager to get range iterator to scan each range on meta range.
- committed.RangeIterator will get graveler.Iterator and unmarshal it to to the committed record.

### CODE REVIEW

File: graveler/committed/range_iterator.go [hunk 1]
```diff
@@ -23,10 +23,12 @@ func (r *RangeIterator) Next() bool {
 		return false
 	}
 	val := r.it.Value()
+	// unmarshal value
 	var v *graveler.Value
 	if val.Value != nil {
 		v, r.err = UnmarshalValue(val.Value)
 		if r.err != nil {
+			r.value = nil
```
Comment (arielshaqed) [inline graveler/committed/range_iterator.go:31]: Actually I don't particularly like this.  If `UnmarshalValue` decided to return a value, why erase it here?

File: graveler/testutil/fakes.go [hunk 2]
```diff
@@ -329,6 +330,44 @@ func (r *valueIteratorFake) Err() error {
...
+	return &committedValueIteratorFake{records: records, current: -1}
+}
+
+func (r *committedValueIteratorFake) Next() bool {
+	r.current++
+	return r.current < len(r.records)
+}
+
+func (r *committedValueIteratorFake) SeekGE(id committed.Key) {
+	for i, record := range r.records {
```
Comment (arielshaqed) [inline graveler/testutil/fakes.go:349]: I think this should `sort.BSearch`.

**Review by arielshaqed — APPROVED** [review]
Thanks!

File: graveler/committed/range_iterator.go (hunk 1 above)
Comment (nopcoder) [reply graveler/committed/range_iterator.go:31]: so .Err() will return the error and .Value() will return nil

**Pull request merged**